frontend/src/app/modules/lessons/lesson/lesson.component.ts

```diff
@@ -71,33 +75,81 @@ export class LessonComponent extends BaseComponent implements OnInit {
         });
     }
 
-    enrollLesson() {
-        this.userService.enrollUserToLesson(this.lesson.id)
-            .pipe(this.untilThis)
-            .subscribe({ next: (lesson) => {
-                this.lesson.isSubscribed = true;
-                this.lesson.subscribersCount = lesson.subscribersCount;
-                this.notificationService.showSuccess(`You successfully registered for lesson ${this.lesson.title}`, 'Success!');
-            },
-            error: () => this.notificationService.showError(`Failed to register for lesson ${this.lesson.title}`, 'Failed!') });
+    buttonAction() {
+        if (this.page === 'timetable') {
+            this.userService
+                .enrollUserToLesson(this.lesson.id)
+                .pipe(this.untilThis)
+                .subscribe({
+                    next: (lesson) => {
+                        this.lesson.isSubscribed = true;
+                        this.lesson.subscribersCount = lesson.subscribersCount;
+                        this.notificationService.showSuccess(
+                            `You successfully registered for lesson ${this.lesson.name}`,
+                            'Success!',
+                        );
+                    },
+                    error: () =>
+                        this.notificationService.showError(
+                            `Failed to register for lesson ${this.lesson.name}`,
+                            'Failed!',
+                        ),
+                });
+        }
+        if (this.page === 'teacher') {
+            this.lessonsService.cancelLesson(this.lesson.id, this.lesson).subscribe((data) => {
+                this.lesson.isCanceled = data.isCanceled;
+            });
+        }
```

Comment:
This way it's better to pass function from parent component as input and execute it